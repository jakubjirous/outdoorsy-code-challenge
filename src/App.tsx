import React from 'react';

function App() {
  return (
    <>
      It works!
    </>
  );
}

export default App;
