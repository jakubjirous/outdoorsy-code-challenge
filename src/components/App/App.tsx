import React from "react";
import styled from "styled-components";
import { deviceScreen, laptopMd, tablet } from "../../utils/responsive";
import Search from "../Search/Search";

const S = {
  Main: styled.main`
    width: 100%;
    height: 100vh;
    display: flex;
    flex-flow: column;
    margin: auto;

    @media screen and ${deviceScreen.tablet} {
      max-width: ${tablet}px;
    }

    @media screen and ${deviceScreen.laptopLg} {
      max-width: ${laptopMd}px;
    }
  `,
  Container: styled.div`
    flex: 1;
    display: flex;
    flex-flow: column;
    gap: var(--spacing-6);
    margin: var(--spacing-5);
  `,
};

function App() {
  return (
    <S.Main>
      <S.Container>
        <Search />
        List <br />
      </S.Container>
    </S.Main>
  );
}

export default App;
