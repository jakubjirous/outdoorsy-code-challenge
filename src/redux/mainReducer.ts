import { combineReducers } from "@reduxjs/toolkit";
import { rentalApi } from "./features/rental/rentalApi";

export const mainReducer = combineReducers({
  // Add the generated reducer as a specific top-level slice
  [rentalApi.reducerPath]: rentalApi.reducer,
});
