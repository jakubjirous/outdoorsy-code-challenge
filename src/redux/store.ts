import { configureStore } from "@reduxjs/toolkit";
import { rentalApi } from "./features/rental/rentalApi";
import { mainReducer } from "./mainReducer";

export const store = configureStore({
  reducer: mainReducer,

  // Adding the api middleware enables caching, invalidation, polling, and other useful features of `rtk-query`.
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(rentalApi.middleware),
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
