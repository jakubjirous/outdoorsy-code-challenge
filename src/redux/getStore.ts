import { configureStore, StoreEnhancer } from "@reduxjs/toolkit";
import { rentalApi } from "./features/rental/rentalApi";
import { mainReducer } from "./mainReducer";
import { RootState, store } from "./store";

/**
 * Gets a store set up with some givens initial state.
 * The store is set up with the RTK Query middleware.
 * Useful mainly for creating redux store provider when writing unit tests.
 *
 * @param initialState
 * @param enhancers
 */
export function getStore(initialState?: RootState, enhancers?: StoreEnhancer[]): typeof store {
  return configureStore({
    reducer: mainReducer,
    middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(rentalApi.middleware),
    enhancers,
  });
}
