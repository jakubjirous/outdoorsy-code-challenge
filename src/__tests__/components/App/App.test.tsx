import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import App from "../../../components/App/App";
import { getStore } from "../../../redux/getStore";

const renderContainer = () => {
  return render(
    <Provider store={getStore()}>
      <App />
    </Provider>
  );
};

describe("<App/>", () => {
  test("should render correctly", () => {
    const { baseElement } = renderContainer();

    expect(baseElement).toMatchSnapshot();
  });
});
