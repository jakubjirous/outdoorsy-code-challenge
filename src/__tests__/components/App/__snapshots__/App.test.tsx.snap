// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should render correctly 1`] = `
<body>
  <div>
    <main
      class="sc-fEOsli bjUOts"
    >
      <div
        class="sc-bjUoiL hSWJwl"
      >
        <header
          class="sc-crXcEl cFwuwv"
        >
          <div
            class="sc-evZas jZsTWM"
          />
          <input
            aria-label="Search"
            class="sc-ksZaOG kMPKML"
            placeholder="Find the perfect RV"
            value=""
          />
        </header>
        <section
          class="sc-kDDrLX fZbxEL"
        >
          <div
            class="sc-jSMfEi beGjEx"
          >
            <div
              class="sc-gKXOVf gnvyZk"
            >
              <div
                class="sc-bczRLJ hATIyy"
                height="8.4375rem"
                width="11.25rem"
              />
              <div
                class="sc-bczRLJ dZTAAT"
                height="var(--fs-lg)"
                width="50%"
              />
            </div>
            <div
              class="sc-gKXOVf gnvyZk"
            >
              <div
                class="sc-bczRLJ hATIyy"
                height="8.4375rem"
                width="11.25rem"
              />
              <div
                class="sc-bczRLJ dZTAAT"
                height="var(--fs-lg)"
                width="50%"
              />
            </div>
            <div
              class="sc-gKXOVf gnvyZk"
            >
              <div
                class="sc-bczRLJ hATIyy"
                height="8.4375rem"
                width="11.25rem"
              />
              <div
                class="sc-bczRLJ dZTAAT"
                height="var(--fs-lg)"
                width="50%"
              />
            </div>
            <div
              class="sc-gKXOVf gnvyZk"
            >
              <div
                class="sc-bczRLJ hATIyy"
                height="8.4375rem"
                width="11.25rem"
              />
              <div
                class="sc-bczRLJ dZTAAT"
                height="var(--fs-lg)"
                width="50%"
              />
            </div>
          </div>
        </section>
      </div>
    </main>
  </div>
</body>
`;
